src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -125,7 +133,7 @@ internal sealed class EvaluationContext : EvaluationContextBase
         {
             // Re-use the previous compilation if possible.
             CSharpCompilation compilation;
-            if (previous.Matches(metadataBlocks))
+            if (previous.Matches(metadataBlocks, aliases))
```

Comment:
We should re-use the compilation if the metadataBlocks match, even if the aliases do not.